src/game/game.cpp

```diff
@@ -10409,6 +10539,403 @@ void Game::playerRewardChestCollect(uint32_t playerId, const Position &pos, uint
 	}
 }
 
+void Game::playerOpenStore(uint32_t playerId) {
+	std::shared_ptr<Player> player = getPlayerByID(playerId);
+	if (!player) {
+		return;
+	}
+
+	if (player->isUIExhausted()) {
+		player->sendCancelMessage(RETURNVALUE_YOUAREEXHAUSTED);
+		return;
+	}
+
+	player->updateUIExhausted();
+	player->openStore();
+}
+
+void Game::playerCoinTransfer(uint32_t playerId, std::string receptorName, uint32_t coinAmount) {
+	std::shared_ptr<Player> playerDonator = getPlayerByID(playerId);
+	if (!playerDonator) {
+		return;
+	}
+
+	if (playerDonator->isUIExhausted(1000)) {
+		playerDonator->sendCancelMessage(RETURNVALUE_YOUAREEXHAUSTED);
+		return;
+	}
+
+	std::shared_ptr<Player> playerReceptor = getPlayerByName(receptorName, true);
+	if (!playerReceptor) {
+		return;
+	}
+
+	if (playerDonator == playerReceptor || playerDonator->getAccountId() == playerReceptor->getAccountId()) {
+		playerDonator->sendStoreError(StoreErrors_t::TRANSFER, "You cannot gift Tibia Coins to characters of your own account.");
+		return;
+	}
+
+	auto [transferableCoins, result] = playerDonator->getAccount()->getCoins(enumToValue(CoinType::Transferable));
+	if (coinAmount > transferableCoins) {
+		playerDonator->sendStoreError(StoreErrors_t::TRANSFER, "You don't have enough coins.");
+		return;
+	}
+
+	std::string historyDesc = fmt::format("{} gifted to {}", playerDonator->getName(), playerReceptor->getName());
+	playerDonator->getAccount()->removeCoins(enumToValue(CoinType::Transferable), coinAmount, historyDesc);
+	playerReceptor->getAccount()->addCoins(enumToValue(CoinType::Transferable), coinAmount, historyDesc);
+
+	StoreHistory tempHistory;
+	tempHistory.description = historyDesc;
+	tempHistory.coinType = enumToValue(CoinType::Transferable);
+	tempHistory.historyType = enumToValue(HistoryTypes_t::NONE);
+	tempHistory.coinAmount = static_cast<int32_t>(coinAmount * -1);
+	tempHistory.createdAt = getTimeNow();
+
+	playerDonator->setStoreHistory(tempHistory);
+
+	if (playerReceptor->isOffline()) {
+		std::shared_ptr<Player> newPlayerReceptor;
+		const auto playersAccountVector = getPlayersByAccount(playerReceptor->getAccount());
+		for (const auto player : playersAccountVector) {
+			if (player->isOnline()) {
+				newPlayerReceptor = player;
+			}
+		}
+
+		if (newPlayerReceptor) {
+			tempHistory.coinAmount = coinAmount;
+			newPlayerReceptor->sendCoinBalance();
+			newPlayerReceptor->setStoreHistory(tempHistory);
+		} else {
+			playerReceptor->getAccount()->registerStoreTransaction(tempHistory.historyType, coinAmount, tempHistory.coinType, historyDesc, tempHistory.createdAt);
+		}
+	} else {
+		tempHistory.coinAmount = coinAmount;
+		playerReceptor->sendCoinBalance();
+		playerReceptor->setStoreHistory(tempHistory);
+	}
```

Comment:
This seems redundant to me, the boolean in: std::shared_ptr<Player> playerReceptor = getPlayerByName(receptorName, true);

It already handles the offline player properly, doesn't it? I think this can be simplified.